server/src/main/java/io/pravega/schemaregistry/server/rest/resources/AbstractResource.java

```diff
@@ -0,0 +1,162 @@
+ */
+package io.pravega.schemaregistry.server.rest.resources;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import io.pravega.auth.AuthException;
+import io.pravega.auth.AuthHandler;
+import io.pravega.auth.AuthenticationException;
+import io.pravega.auth.AuthorizationException;
+import io.pravega.common.concurrent.Futures;
+import io.pravega.schemaregistry.server.rest.ServiceConfig;
+import io.pravega.schemaregistry.server.rest.auth.AuthHandlerManager;
+import io.pravega.schemaregistry.service.SchemaRegistryService;
+import lombok.Getter;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.curator.shaded.com.google.common.base.Charsets;
+
+import javax.ws.rs.container.AsyncResponse;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.Response;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+import java.util.function.Supplier;
+
+import static io.pravega.schemaregistry.server.rest.resources.AuthResources.*;
+
+@Slf4j
+abstract class AbstractResource {
+    @Context
+    HttpHeaders headers;
+
+    @Getter
+    private final SchemaRegistryService registryService;
+    @Getter
+    private final ServiceConfig config;
+    @Getter
+    private final AuthHandlerManager authManager;
+    @Getter
+    private final Executor executorService;
+
+    AbstractResource(SchemaRegistryService registryService, ServiceConfig config, Executor executorService) {
+        this.registryService = registryService;
+        this.config = config;
+        this.authManager = new AuthHandlerManager(config);
+        this.executorService = executorService;
+    }
+
+    /**
+     * This is a shortcut for {@code headers.getRequestHeader().get(HttpHeaders.AUTHORIZATION)}.
+     *
+     * @return a list of read-only values of the HTTP Authorization header
+     * @throws IllegalStateException if called outside the scope of the HTTP request
+     */
+    List<String> getAuthorizationHeader() {
+        return headers.getRequestHeader(HttpHeaders.AUTHORIZATION);
+    }
+
+    CompletableFuture<Response> withCompletion(String request, AuthHandler.Permissions permissions,
+                                               String resource, AsyncResponse response,
+                                               Supplier<CompletableFuture<Response>> future) {
+        try {
+            authenticateAuthorize(getAuthorizationHeader(), resource, permissions);
+            return future.get();
+        } catch (AuthException e) {
+            log.warn("Auth failed {}", request, e);
+            response.resume(Response.status(Response.Status.fromStatusCode(e.getResponseCode())).build());
+            throw e;
+        } catch (IllegalArgumentException e) {
+            log.warn("Bad request {} error:{}", request, e.getMessage());
```

Comment:
nit: `Bad request {} error:{}` -> `Illegal request {}. Error: {}`

`Bad request` may be misleading here.

Reply:
Bad request is because the HTTP error code being returned is BadRequest.